custom/aaa/views.py

```diff
@@ -46,18 +47,34 @@ def domain(self):
     def couch_user(self):
         return self.request.couch_user
 
+    @property
+    def user_ministry(self):
+        return self.couch_user.user_data.get('ministry')
+
+    def dispatch(self, *args, **kwargs):
+        if (not self.couch_user.is_web_user()
+                and (self.user_ministry is None or self.user_ministry == '')):
+            return no_permissions(self.request)
+
+        return super(ReachDashboardView, self).dispatch(*args, **kwargs)
+
     def get_context_data(self, **kwargs):
         kwargs['domain'] = self.domain
-        # TODO add logic for user role type possible options should be MoHFW or MWCD
-        kwargs['user_role_type'] = 'MoHFW'
+
+        kwargs['is_web_user'] = self.couch_user.is_web_user()
+        kwargs['user_role_type'] = self.user_ministry
+
+        if (not kwargs['is_web_user']
+                and (kwargs['user_role_type'] is None or kwargs['user_role_type'] == '')):
+            return no_permissions(self.request)
```

Comment:
This one could also use a `message`

Reply:
surely this could never be triggered since the same check is done in `dispatch`

Reply:
That's correct and I removed it https://github.com/dimagi/commcare-hq/pull/23459/commits/7565d015fd559677d4674f79abf50a3168b131a5